### DISCUSSION TITLE
FirebaseAuthentication addListener does not listen

### DISCUSSION THREAD

**Discussion by FrancescoIoppolo** [capawesome-team/capacitor-firebase#532] [Q&A] [answered]
Good evening and happy holidays everyone!  🎉🎊🥳 
I'm trying to find a method to listen to a user's changes in fierbase.
This is my approach in reactjs

```
  useEffect(() => {
    const initializeListener = async () => {
      try {
        FirebaseAuthentication.addListener('authStateChange', (result) => {
          console.log(result.user || undefined);
        });
      } catch (error) {
        console.error('Errore durante l\'inizializzazione di Firebase:', error);
      }
    };

    initializeListener();
  }, []);
```

I haven't found a real example to do this, but unfortunately I can't keep listening.
Example: If I delete the user from inside the firebase console nothing happens in my app.

I state that when I load the component it returns the user object so there are no project configuration problems

Thanks for your support! 🙏🙏

**Comment by robingenz** [comment, marked as answer]
> I'm trying to find a method to listen to a user's changes in fierbase.

Your approach is correct. `addListener('authStateChange', ...)` is the right method.
**Attention**: If you also use the Firebase JS SDK on Android and iOS, then you must use the listener from the Firebase JS SDK.

>  Example: If I delete the user from inside the firebase console nothing happens in my app.

Yes, this is because the Firebase SDK does not call this listener in this case, see this [SO answer](https://stackoverflow.com/a/38354484/6731412).

**Comment by FrancescoIoppolo** [comment]
OK thank you! 
Everything very clear. 
now I wonder: Does it work when a user verifies his email? Otherwise how can I update this event in the database when the password is verified? do you have any idea what the best approach might be?

**Reply by robingenz** [reply]
> Does it work when a user verifies his email?

Give it a try. I don't know right now. ^^

**Comment by jonasermert** [comment]
```
useEffect(() => {
  const initializeListener = async () => {
    try {
      FirebaseAuthentication.onAuthStateChanged((user) => {
        console.log(user || undefined);
      });
    } catch (error) {
      console.error('Error during Firebase initialization:', error);
    }
  };

  initializeListener();
}, []);

```
